src/bh_route_accounts.erl

```diff
@@ -14,55 +14,89 @@
+                            [?SELECT_ACCOUNT_BASE,
+                             "where (l.address > $1 and l.first_block = $2) or (l.first_block < $2) ",
+                             "order by first_block desc, address limit ", integer_to_list(?ACCOUNT_LIST_LIMIT)],
+                            []),
 
     {ok, S2} = epgsql:parse(Conn, ?S_ACCOUNT_LIST,
-                           ?SELECT_ACCOUNT_BASE "order by first_block desc, address limit $1", []),
+                             [?SELECT_ACCOUNT_BASE,
+                              "order by first_block desc, address limit ", integer_to_list(?ACCOUNT_LIST_LIMIT)],
+                             []),
 
     {ok, S3} = epgsql:parse(Conn, ?S_ACCOUNT,
-                           ?SELECT_ACCOUNT_BASE(
+                           [?SELECT_ACCOUNT_BASE(
                               ", (select coalesce(max(nonce), l.nonce) from pending_transactions p where p.address = l.address and nonce_type='balance' and status != 'failed') as speculative_nonce"
-                             )
-                           "where l.address = $1", []),
+                             ), "where l.address = $1"],
+                            []),
 
-    #{?S_ACCOUNT_LIST_BEFORE => S1, ?S_ACCOUNT_LIST => S2,
-      ?S_ACCOUNT => S3}.
+    #{?S_ACCOUNT_LIST_BEFORE => S1,
+      ?S_ACCOUNT_LIST => S2,
+      ?S_ACCOUNT => S3 }.
 
 handle('GET', [], Req) ->
-    Args = ?GET_ARGS([before, limit], Req),
+    Args = ?GET_ARGS([cursor], Req),
     ?MK_RESPONSE(get_account_list(Args));
 handle('GET', [Account], _Req) ->
     ?MK_RESPONSE(get_account(Account));
 handle('GET', [Account, <<"hotspots">>], Req) ->
-    Args = ?GET_ARGS([before, limit], Req),
+    Args = ?GET_ARGS([cursor], Req),
     ?MK_RESPONSE(bh_route_hotspots:get_hotspot_list([{owner, Account} | Args]));
-handle('GET', [Account, <<"transactions">>], Req) ->
-    Args = ?GET_ARGS([filter_types, before, limit], Req),
-    ?MK_RESPONSE(bh_route_txns:get_txn_list([{actor, Account} | Args]));
 
 handle(_, _, _Req) ->
     ?RESPONSE_404.
 
-get_account_list([{before, undefined}, {limit, Limit}])  ->
-    {ok, _, Results} = ?PREPARED_QUERY(?S_ACCOUNT_LIST, [Limit]),
-    {ok, account_list_to_json(Results)};
-get_account_list([{before, Before}, {limit, Limit}]) ->
-    {ok, _, Results} = ?PREPARED_QUERY(?S_ACCOUNT_LIST_BEFORE, [Before, Limit]),
-    {ok, account_list_to_json(Results)}.
+get_account_list([{cursor, undefined}])  ->
+    Result = ?PREPARED_QUERY(?S_ACCOUNT_LIST, []),
+    mk_account_list_from_result(undefined, Result);
+get_account_list([{cursor, Cursor}]) ->
+    case ?CURSOR_DECODE(Cursor) of
+        {ok, #{ <<"before_address">> := BeforeAddress,
+                <<"before_block">> := BeforeBlock,
+                <<"height">> := CursorHeight}} ->
+            Result = ?PREPARED_QUERY(?S_ACCOUNT_LIST_BEFORE, [BeforeAddress, BeforeBlock]),
+            mk_account_list_from_result(CursorHeight, Result);
+        {error, _} ->
```

Comment:
You might want to just make this _ or a malformed cursor (that is still base64'd json) will case_clause here

Reply:
https://github.com/helium/blockchain-http/pull/30/commits/7858bb58ae5e51173b50f8fc898f723acc2c2b2e